core/src/main/scala/kafka/zk/KafkaZkClient.scala

```
@@ -539,6 +543,176 @@ class KafkaZkClient(zooKeeperClient: ZooKeeperClient, isSecure: Boolean) extends
     retryRequestsUntilConnected(deleteRequests)
   }
 
+  //Acl management methods
+
+  /**
+   * Creates the required zk nodes for Acl storage
+   */
+  def createAclPaths(): Unit = {
+    createRecursive(AclZNode.path)
+    createRecursive(AclChangeNotificationZNode.path)
+    ResourceType.values.foreach(resource => createRecursive(ResourceTypeZNode.path(resource.name)))
+  }
+
+  /**
+   * Gets VersionedAcls for a given Resource
+   * @param resource Resource to get VersionedAcls for
+   * @return  VersionedAcls
+   */
+  def getVersionedAclsForResource(resource: Resource): VersionedAcls = {
+    val getDataRequest = GetDataRequest(ResourceZNode.path(resource))
+    val getDataResponse = retryRequestUntilConnected(getDataRequest)
+    if (getDataResponse.resultCode == Code.OK) {
+      ResourceZNode.decode(getDataResponse.data, getDataResponse.stat)
+    } else {
+      VersionedAcls(Set(), -1)
+    }
+  }
+
+  /**
+   * Sets Resource znode with the given acls.
+   * @param resource  resource object
+   * @param acls the acls to sets on the resource znode
+   * @param expectedVersion expected zk version
+   * @return SetDataResponse
+   */
+  def setAclForResource(resource: Resource, acls: Set[Acl], expectedVersion: Int): SetDataResponse = {
+    val setDataRequest = SetDataRequest(ResourceZNode.path(resource), ResourceZNode.encode(acls), expectedVersion)
+    retryRequestUntilConnected(setDataRequest)
+  }
+
+  /**
+   * Creates Resource znode with the given acls
+   * @param resource  resource object
+   * @param aclsSet the acls to sets on the resource znode
+   * @return CreateResponse
+   */
+  def createAclForResource(resource: Resource, aclsSet: Set[Acl]): CreateResponse = {
+    val path = ResourceZNode.path(resource)
+    val createRequest = CreateRequest(path, ResourceZNode.encode(aclsSet), acls(path), CreateMode.PERSISTENT)
+    retryRequestUntilConnected(createRequest)
```

Since the caller is not checking the response code, we probably want to throw an exception if result code is not OK and change the return type of the method to Unit.

---

In this case, caller (SimpleAclAuthorizer) is checking the response code.  updated the code related to createAclChangeNotificationRaw method.